code/modules/events/crystal_event.dm

```diff
@@ -48,8 +48,19 @@ This section is for the event controller
 	var/obj/machinery/destabilized_crystal/dest_crystal
 	///Check if the event will end badly
 	var/is_zk_scenario = TRUE
+	///Ticks that have to pass before second wave
+	var/second_wave = 310
 
 /datum/round_event/crystal_invasion/start()
+	var/list/sm_crystal = list()
+	for(var/obj/machinery/power/supermatter_crystal/temp in GLOB.machines)
+		if(istype(temp, /obj/machinery/power/supermatter_crystal/shard))
+			continue
+		sm_crystal += temp
+	if(sm_crystal == null)
```

Comment:
```suggestion
	if(!length(sm_crystal))
```
This check would have never failed, since you defined the variable as a list, it cannot be `null`.